data/spells/lib/spells.lua

```diff
@@ -296,3 +306,74 @@ function Creature:addDamageCondition(target, conditionType, listType, damage, ti
 	target:addCondition(condition)
 	return true
 end
+
+function isOnCombatArea(casterPosition, targetPosition, coordinates)
+	if casterPosition.z == targetPosition.z then
+		for _, axis in ipairs(coordinates) do
+			if casterPosition.x + axis.x == targetPosition.x and casterPosition.y + axis.y == targetPosition.y then
+				return true
+			elseif casterPosition.x - axis.x == targetPosition.x and casterPosition.y + axis.y == targetPosition.y then
+				return true
+			elseif casterPosition.x - axis.x == targetPosition.x and casterPosition.y - axis.y == targetPosition.y then
+				return true
+			elseif casterPosition.x + axis.x == targetPosition.x and casterPosition.y - axis.y == targetPosition.y then
+				return true
+			end
+		end
+	end
+	return false
+end
```

Comment:
I thought I could get rid of this bug just by using Cartesian coordenates (that's why I had to create that table) and checking if a party member was on any square of the combat area (I just added the first quadrant of the original area to that table because all others Q's were symmetric) and also checking if the member was on the same floor level.

I thought about using tile, target combat callbacks but that wouldn't work because of the *required mana formula* which is based on number of affected members (the formula must be used only once to get the correct value.)